core/store/orm/orm.go

```diff
@@ -326,6 +339,44 @@ func (orm *ORM) LinkEarnedFor(spec *models.JobSpec) (*assets.Link, error) {
 	return earned, nil
 }
 
+// UpsertErrorFor upserts a JobSpecError record, incrementing the occurences counter by 1
+// if the record is found
+func (orm *ORM) UpsertErrorFor(jobID *models.ID, description string) {
```

Comment:
It would be less surprised to have this return an `error` then log it in the parent

Reply:
Discussed this with @se3000 above. Returning error was my original implementation but it's a bit strange to ask the parent to handle the insert error the parent is already in an error handling code path.

Reply:
Fair enough